lib/Class/Mock/Generic/InterfaceTester.pm

```diff
@@ -84,6 +113,50 @@ that code-ref for validation.  It should return true if the params are OK
 and false otherwise.  In the example, it will return true if the hash of
 args contains a 'fruit' key with value 'apple'.
 
+=head2 add_fixtures
+
+ In: \@fixtures or @fixtures
+
+Supplied with either an arrayref or a list of method call fixtures, adds them
+to the array of fixtures this object maintains internally (although see below
+for a caveat about this).
+
+At the simplest, a method call fixture is a hashref with keys
+C<method>, C<input> and C<output>. If you don't care about the input
+your method receives, you can omit that key and any input will be accepted.
+
+You can also provide a fixture as a pair of C<method> and (hashref containing
+input and output). This lets you write a series of method call fixtures as an
+apparent ordered hash, which may feel more natural. As above, you can omit
+the input field if you don't care. So the following calls are equivalent:
+
+ $interface_tester->add_fixtures(
+     [
+         {
+             method => 'do_something',
+             input  => sub { 1 },
+             output => 'Yup, done',
+         },
+         {
+             method => 'do_something_with_this',
+             input  => ['fish'],
+             output => 'Fish cooked',
+         }
+     ]
+ );
+
+ $interface_tester->add_fixtures(
+     do_something           => { output => 'Yup, done' },
+     do_something_with_this => {
+         input  => ['fish'],
+         output => 'Fish cooked',
+     },
+ );
+
+Caveat: just in case you need to test a call to a method that coincidentally
+is also called C<add_fixtures>, this method is only enabled
+if you did I<not> provide a list of fixtures to the constructor.
```

Comment:
That, to me, is somewhat surprising behaviour; I think it'd be better to just clearly document that "you can't mock a method named `add_fixtures` because we already use it, so tough".  Otherwise, it would be too easy to be extending an existing test where someone defined their mocks up front, and use `add_fixtures` to add your own in your test (perhaps meaning to refactor the rest later, but not having time), and be bitten by it.

If you're trying to mock `add_fixtures`, then you're probably trying to mock mocking, or something, and you're doing it wrong :)